verl/workers/reward_manager/generative.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2024 PRIME team and/or its affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+from concurrent.futures import ProcessPoolExecutor
+from functools import partial
+from typing import Callable, Optional
+
+import torch
+from transformers import PreTrainedTokenizer
+
+from verl import DataProto
+from verl.utils.reward_score.generative import process_data_async
+from verl.utils.reward_score import _default_compute_score
+import yaml
+
+
+def _compute_score(data_source, solution_str, ground_truth,
+                   extra_info, config) -> torch.Tensor:
+    loop = asyncio.get_event_loop()
+    return loop.run_until_complete(process_data_async(data_source, solution_str, ground_truth, extra_info, config))
+
+
+class GenerativeRewardManager:
+
```

Comment:
please add docstring to this new class

Reply:
could you add a unit test for this class, construct a manager, and invoke it?
additionally if you have end-to-end example that shows the implementation works for a RL dataset it would be great